File: src/org/antlr/runtime/MismatchedTreeNodeException.java
package org.antlr.runtime;

import org.antlr.runtime.tree.TreeNodeStream;
import org.antlr.runtime.tree.Tree;

/**
 */
public class MismatchedTreeNodeException extends RecognitionException {
	public int expecting;
	public Object foundNode;

	public MismatchedTreeNodeException() {
	}

	public MismatchedTreeNodeException(int expecting, TreeNodeStream input) {
		super(input);
		if ( input.LT(1) instanceof Tree ) {
			Tree t = (Tree)input.LT(1);
			line = t.getLine();
			charPositionInLine = t.getCharPositionInLine();
			// TODO: if DOWN/UP, there is no line info currently
		}
		this.foundNode = input.LT(1);
		this.expecting = expecting;
	}

	public String toString() {
		return "MismatchedTreeNodeException("+getUnexpectedType()+"!="+expecting+")";
	}
}

File: runtime/CSharp/Antlr3.Runtime/Antlr.Runtime/MismatchedTreeNodeException.cs
namespace Antlr.Runtime
{
    using Antlr.Runtime.Tree;
    using System;

    [Serializable]
    public class MismatchedTreeNodeException : RecognitionException
    {
        public int expecting;

        public MismatchedTreeNodeException()
        {
        }

        public MismatchedTreeNodeException(int expecting, ITreeNodeStream input) : base(input)
        {
            ITree tree = (ITree) input.LT(1);
            if (input.LT(1) is ITree)
            {
                base.Line = tree.Line;
                base.CharPositionInLine = tree.CharPositionInLine;
            }
            this.expecting = expecting;
        }

        public override string ToString()
        {
            return string.Concat(new object[] { "MismatchedTreeNodeException(", this.UnexpectedType, "!=", this.expecting, ")" });
        }
    }
}

